linktimeplugin.hpp

```diff
@@ -59,32 +53,33 @@ namespace linktimeplugin {
         // Implemented by the derived registrar class.
         virtual BASE& operator()() = 0;
 
+        typedef std::vector<BASE*> PluginColl;
+        typedef std::vector<RegistrarBase<BASE>*> RegistarColl;
+        
         // Returns all registrars.
-        static std::vector<BASE*> plugins() {
-            std::vector<BASE*> ret;
-
-            if (registrars_) {
-                for(auto r : *registrars_) {
-                    ret.push_back(&(*r)());
-                }
+        static PluginColl plugins() {
+            PluginColl ret;
+            for(auto r : registrars()) {
+                ret.push_back(&(*r)());
             }
-
             return ret;
         }
-
+        // Use (*begin())() to get BASE object!
+        static auto begin() {
```

Comment:
Is this use of auto supported in C++11 already?

Reply:
My bad. Haven't considered C++11. Enjoyed C++17 given in the cmake file. I should have remembered the comment was saying it didn't really needed C++17.

Reply:
My bad as well because I forgot to change from 17 to 11 in the cmake file. This was originally written for a C++17 application, and I never bothered to test it on older versions until the requirement to use it in 11 came up recently.